src/interface/blas2/backend/default.hpp

```diff
@@ -137,8 +137,21 @@ typename sb_handle_t::event_t _trsv(
     sb_handle_t& sb_handle, index_t _N, container_t0 _mA, index_t _lda,
     container_t1 _vx, increment_t _incx,
     typename sb_handle_t::event_t _dependencies) {
-  return blas::internal::_trsv_impl<4, 2, uplo, trn, diag>(
-      sb_handle, _N, _mA, _lda, _vx, _incx, _dependencies);
+  const auto device = sb_handle.get_queue().get_device();
+  const std::string vendor =
+      device.template get_info<sycl::info::device::vendor>();
+  if (device.is_gpu()) {
+    if (vendor.find("Intel") == vendor.npos) {
+      return blas::internal::_trsv_impl<32, 4, uplo, trn, diag>(
```

Comment:
Why does the tuning of `trsv_impl` change here?

Reply:
All the if-statement you see are there to change those templates parameters. The first one set the `subgroup` size used inside the kernel and it used also to set localSize and globalSize  to launch the kernel. Without being certain that this parameter are a multiple of the actual subgroup size on the hardware `txsv` kernels hang on NVIDIA due to group operations like broadcast.
The second template parameter changed is the number of subgroups of the given subgroup size. It can be different but at this point I used the same value of the proper tuned kernel, because it gave the best performance when benchmarked while adding these operators to portBLAS.
Same reason applies to `tpsv` and `tbsv`.